src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -397,6 +397,31 @@ public void RegisterStreamProvider(string providerTypeFullName, string providerN
             ProviderConfigurationUtility.RegisterProvider(ProviderConfigurations, ProviderCategoryConfiguration.STREAM_PROVIDER_CATEGORY_NAME, providerTypeFullName, providerName, properties);
         }
 
+
+        public void RegisterStatisticsProvider<T>(string providerName, IDictionary<string, string> properties = null) where T : IStatisticsPublisher, IClientMetricsDataPublisher
+        {
+            TypeInfo providerTypeInfo = typeof(T).GetTypeInfo();
+            if (providerTypeInfo.IsAbstract ||
+                providerTypeInfo.IsGenericType ||
+                providerTypeInfo.IsGenericType ||
+                !(
+                typeof(IStatisticsPublisher).IsAssignableFrom(providerTypeInfo) &&
+                typeof(IClientMetricsDataPublisher).IsAssignableFrom(providerTypeInfo)
+                ))
```

Comment:
Your build in netstandard environment failed on 

> \src\Orleans\Configuration\ClientConfiguration.cs(408,63): error CS1503: Argument 2: cannot convert from 'System.Reflection.TypeInfo' to 'System.Type' 

Because in netstandard, `TypeInfo `doesn't Inherit `Type` class anymore. And signature of `IsAssignableFrom` is 
`Type.IsAssignableFrom(System.Type).` You need to do 
TypeproviderType = typeof(T).GetType();